src/oge/validation.py

```diff
@@ -52,64 +52,98 @@ def check_allocated_gf_matches_input_gf(year, gen_fuel_allocated):
+        )
+        if len(input_minus_allocated) > 0:
+            logger.warning(
+                f"There are {len(input_minus_allocated)} plant/fuel combinations in "
+                "EIA-923 Input but not in allocated EIA-923 : \n"
+                + plant_total_gf.loc[input_minus_allocated]
+                .assign(ba_code=[id2ba[i] for i, _ in input_minus_allocated])
+                .to_string()
+            )
+        allocated_minus_input = list(
+            set(plant_total_alloc.index).difference(set(plant_total_gf.index))
+        )
+        if len(allocated_minus_input) > 0:
+            logger.warning(
+                f"There are {len(allocated_minus_input)} plant/fuel combinations in "
+                "allocated EIA-923 but not in EIA-923 Input: \n"
+                + plant_total_alloc.loc[allocated_minus_input]
+                .assign(ba_code=[id2ba[i] for i, _ in allocated_minus_input])
+                .to_string()
+            )
+
+    # Align the 2 data frames. Values in missing rows in either data frames are set to 0.
+    plant_total_gf, plant_total_alloc = plant_total_gf.align(
+        plant_total_alloc, fill_value=0
+    )
+    # Calculate the percentage difference between the values
+    # Replace 0s with small sentinel value to prevent missing values from divide by zero
+    plant_total_diff = (plant_total_alloc - plant_total_gf) / plant_total_gf.replace(
+        0, 0.00001
     ).dropna(how="all", axis=0)
-    # flag rows where the absolute percentage difference is greater than our threshold
+    # Flag rows where the absolute percentage difference is greater than our threshold
     mismatched_allocation = plant_total_diff[
         (~np.isclose(plant_total_diff["fuel_consumed_mmbtu"], 0))
         | (~np.isclose(plant_total_diff["net_generation_mwh"], 0))
     ]
+
     if len(mismatched_allocation) > 0:
-        logger.warning("Allocated EIA-923 doesn't match input data for plants:")
+        logger.warning(
+            f"There are {len(mismatched_allocation)} plant/fuel combinations "
+            "with non-zero net generation or fuel consumed that are different "
+            "in aloocated EIA-923 and EIA-923 Input"
+        )
+        mismatched_ba_code = [
+            id2ba[i]
+            for i in mismatched_allocation.index.get_level_values("plant_id_eia")
+        ]
         logger.warning("Percentage Difference:")
         logger.warning(
-            "\n"
-            + mismatched_allocation.merge(
-                create_plant_ba_table(year)[["plant_id_eia", "ba_code"]],
-                how="left",
-                on="plant_id_eia",
-                validate="m:1",
-            ).to_string()
+            "\n" + mismatched_allocation.assign(ba_code=mismatched_ba_code).to_string()
         )
         logger.warning("EIA-923 Input Totals:")
         logger.warning(
             "\n"
             + plant_total_gf.loc[mismatched_allocation.index, :]
```

Comment:
When we print these input and outputs, we'd want to be able to see where the values are Nas vs zeros, which is why I'm proposing that we don't fill na with zero when aligning the dataframes and only do it when calculating the percent difference.